yarpl/include/yarpl/observable/Observable.h

```diff
@@ -167,6 +203,57 @@ Reference<Observable<T>> Observable<T>::subscribeOn(folly::Executor& executor) {
 }
 
 template <typename T>
+template <typename Function>
+Reference<Observable<T>> Observable<T>::doOnSubscribe(Function function) {
+  return details::createDoOperator(ref_from_this(this), std::move(function), [](const T&){}, [](auto){}, []{});
+}
+
+template <typename T>
+template <typename Function>
+Reference<Observable<T>> Observable<T>::doOnNext(Function function) {
+  return details::createDoOperator(ref_from_this(this), []{}, std::move(function), [](auto){}, []{});
+}
+
+template <typename T>
+template <typename Function>
+Reference<Observable<T>> Observable<T>::doOnError(Function function) {
+  return details::createDoOperator(ref_from_this(this), []{}, [](const T&){}, std::move(function), []{});
+}
+
+template <typename T>
+template <typename Function>
+Reference<Observable<T>> Observable<T>::doOnComplete(Function function) {
+  return details::createDoOperator(ref_from_this(this), []{}, [](const T&){}, [](auto){}, std::move(function));
+}
+
+template <typename T>
+template <typename Function>
+Reference<Observable<T>> Observable<T>::doOnTerminate(Function function) {
+  auto sharedFunction = std::make_shared<Function>(std::move(function));
+  return details::createDoOperator(ref_from_this(this), []{}, [](const T&){}, [sharedFunction](auto){(*sharedFunction)();}, [sharedFunction](){(*sharedFunction)();});
+
+}
+
+template <typename T>
+template <typename Function>
+Reference<Observable<T>> Observable<T>::doOnEach(Function function) {
+  auto sharedFunction = std::make_shared<Function>(std::move(function));
+  return details::createDoOperator(ref_from_this(this), []{}, [sharedFunction](const T&){(*sharedFunction)();}, [sharedFunction](auto){(*sharedFunction)();}, [sharedFunction](){(*sharedFunction)();});
```

Comment:
Why not also provide the lambda for `onSubscribe`?

Reply:
bacause doOnEach operator is specified only for on{N,C,E}